File: app/src/main/java/com/UIN/Tool/plugin/PluginWebChromeClient.java
package com.UIN.Tool.plugin;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.webkit.JsResult;
import android.webkit.WebChromeClient;
import android.webkit.WebView;
import android.widget.Toast;

import com.UIN.Tool.utils.LogUtils;

public class PluginWebChromeClient extends WebChromeClient {
    
    private final Context context;
    private final String pluginId;
    
    public PluginWebChromeClient(Context context, String pluginId) {
        this.context = context;
        this.pluginId = pluginId;
    }
    
    @Override
    public void onProgressChanged(WebView view, int newProgress) {
        super.onProgressChanged(view, newProgress);
        if (newProgress == 100) {
            LogUtils.success("PluginWebChromeClient", "插件加载完成: " + pluginId);
        }
    }
    
    @Override
    public void onReceivedTitle(WebView view, String title) {
        super.onReceivedTitle(view, title);
        LogUtils.i("PluginWebChromeClient", "[" + pluginId + "] 标题: " + title);
    }
    
    @Override
    public boolean onJsAlert(WebView view, String url, String message, JsResult result) {
        LogUtils.i("PluginWebChromeClient", "JS Alert: " + message);
        Toast.makeText(context, message, Toast.LENGTH_LONG).show();
        result.confirm();
        return true;
    }
    
    @Override
    public boolean onJsConfirm(WebView view, String url, String message, JsResult result) {
        LogUtils.i("PluginWebChromeClient", "JS Confirm: " + message);
        // 这里可以显示对话框
        result.confirm();
        return true;
    }
    
    @Override
    public boolean onConsoleMessage(android.webkit.ConsoleMessage consoleMessage) {
        LogUtils.d("WebView[" + pluginId + "]", 
            consoleMessage.message() + " (line " + consoleMessage.lineNumber() + ")");
        return true;
    }
}

File: app/src/main/java/com/UIN/Tool/core/plugin/PluginWebChromeClient.kt
package com.UIN.Tool.core.plugin

import android.content.Context
import android.webkit.JsResult
import android.webkit.WebChromeClient
import android.webkit.WebView
import android.widget.Toast
import com.UIN.Tool.log.Logger

class PluginWebChromeClient(
    private val context: Context,
    private val pluginId: String
) : WebChromeClient() {
    
    private val TAG = "PluginWebChromeClient"
    
    override fun onProgressChanged(view: WebView, newProgress: Int) {
        super.onProgressChanged(view, newProgress)
        if (newProgress == 100) {
            Logger.success(TAG, "插件加载完成: $pluginId")
        }
    }
    
    override fun onReceivedTitle(view: WebView, title: String) {
        super.onReceivedTitle(view, title)
        Logger.i(TAG, "[$pluginId] 标题: $title")
    }
    
    override fun onJsAlert(view: WebView, url: String, message: String, result: JsResult): Boolean {
        Logger.i(TAG, "JS Alert: $message")
        Toast.makeText(context, message, Toast.LENGTH_LONG).show()
        result.confirm()
        return true
    }
    
    override fun onJsConfirm(view: WebView, url: String, message: String, result: JsResult): Boolean {
        Logger.i(TAG, "JS Confirm: $message")
        android.app.AlertDialog.Builder(context)
            .setTitle("确认")
            .setMessage(message)
            .setPositiveButton("确定") { _, _ -> result.confirm() }
            .setNegativeButton("取消") { _, _ -> result.cancel() }
            .show()
        return true
    }
    
    override fun onConsoleMessage(consoleMessage: android.webkit.ConsoleMessage): Boolean {
        Logger.d("WebView[$pluginId]", 
            "${consoleMessage.message()} (line ${consoleMessage.lineNumber()})")
        return true
    }
}